Explorer/Assets/DCL/Infrastructure/ECS/SceneLifeCycle/IncreasingRadius/ResolveSceneStateByIncreasingRadiusSystem.cs

```diff
@@ -96,117 +161,268 @@ private void ProcessScenesUnloadingInRealm()
-
-                    if (partitionComponent.RawSqrDistance >= maxLoadingSqrDistance) continue;
+                    OrderedDataManaged currentOrderedData = orderedDataManaged[i];
+                    currentOrderedData.UpdatePlayerInParcel(xCoordinate, yCoordinate);
 
-                    orderedData.Add(new OrderedData
+                    dataPtr[i] = new OrderedDataNative
                     {
-                        Entity = entity,
-                        Data = new DistanceBasedComparer.DataSurrogate(partitionComponent.RawSqrDistance, partitionComponent.IsBehind),
-                    });
+                        ReferenceListIndex = i,
+                        RawSqrDistance = currentOrderedData.RawSqrDistance,
+                        IsBehind = currentOrderedData.IsBehind,
+                        IsPlayerInsideParcel = currentOrderedData.IsPlayerInsideParcel,
+                        XCoordinate = currentOrderedData.XCoordinate,
+                        OutOfRange = currentOrderedData.OutOfRange,
+                    };
                 }
             }
 
-            // Raw Distance will give more stable results in terms of scenes loading order, especially in cases
-            // when a wide range falls into the same bucket
-            sortingJobHandle = orderedData.SortJob(COMPARER_INSTANCE).Schedule();
+            arraysInSync = true;
+            sortingJobHandle = orderedDataNative.SortJob(COMPARER_INSTANCE).Schedule();
         }
 
         private void CreatePromisesFromOrderedData(IIpfsRealm ipfsRealm)
         {
-            var promisesCreated = 0;
+            loadedScenes = 0;
+            loadedLODs = 0;
+            qualityReductedLOD = 0;
+            promisesCreated = 0;
 
-            for (var i = 0; i < orderedData.Length && promisesCreated < realmPartitionSettings.ScenesRequestBatchSize; i++)
+            TeleportUtils.PlayerTeleportingState teleportParcel = TeleportUtils.GetTeleportParcel(World, playerEntity);
+            //Dont do anything until the teleport cooldown is over
+            if (teleportParcel.JustTeleported)
+                return;
+
+            unsafe
             {
-                OrderedData data = orderedData[i];
+                int orderedDataNativeLength = orderedDataNative.Length;
+                OrderedDataNative* dataPtr = orderedDataNative.GetUnsafePtr();
+                int xCoordinateTeleporting = teleportParcel.Parcel.x;
+                int yCoordinateTeleporting = teleportParcel.Parcel.y;
 
-                // As sorting is throttled Entity might gone out of scope
-                if (!World.IsAlive(data.Entity))
-                    continue;
+                for (var i = 0; i < orderedDataNativeLength && promisesCreated < realmPartitionSettings.ScenesRequestBatchSize; i++)
+                {
+                    OrderedDataManaged data = orderedDataManaged[dataPtr[i].ReferenceListIndex];
```

Comment:
I believe `dataPtr[i].ReferenceListIndex` = `i`, is not it?

Reply:
It is not, same answer as above

Reply:
got it